src/main/java/core/basesyntax/strategy/OperationStrategy.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.Map;
+
+public class OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategy(Map<FruitTransaction.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    public void execute(FruitTransaction.Operation operation,
+                        List<FruitTransaction> transactions,
+                        Map<String, Integer> inventory) {
+        operationHandlers.get(operation).handle(transactions, inventory);
```

Comment:
Consider adding a null check for the result of `operationHandlers.get(operation)` to prevent potential `NullPointerException` if an unsupported operation is passed. You might want to throw a more descriptive exception if the operation is not found.